src/App.tsx

```diff
@@ -1,27 +1,88 @@
-import React from 'react';
+import React, { useCallback, useState } from 'react';
+
+import Button from 'react-bootstrap/Button';
+import Spinner from 'react-bootstrap/Spinner';
+
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+
+import { getAll, get5First, getRedGoods } from './api/goods';
+
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [hasError, setHasError] = useState(false);
+
+  const handleGoodsLoading = useCallback(
+    async (
+      getGoods: () => Promise<Good[]>,
+    ) => {
+      setHasError(false);
+      setIsLoading(true);
+
+      const goodsFromServer = await getGoods()
+        .catch(() => {
+          setHasError(true);
+        })
+        .finally(() => setIsLoading(false));
+
+      setGoods(goodsFromServer || []);
+    },
+    [],
+  );
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+  return (
+    <div className="App">
+      <div className="header">
+        <h1 className="page-header">Dynamic list of Goods</h1>
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+        <div className="buttons-container">
+          <Button
+            type="button"
+            data-cy="all-button"
+            variant="info"
+            className="load-button"
+            onClick={() => handleGoodsLoading(getAll)}
```

Comment:
Prevent clicking the same button twice. If the user has all goods it doesn't need to be loaded from the server again.